### PR TITLE
fix issue Credential was deleted due to "Failed to update online status" #217

### PR DESCRIPTION
**Pull request by qibobo**

### CODE REVIEW

**Comment by qibobo** [comment]
@JohnStarich any comment?

**Comment by JohnStarich** [comment]
@qibobo Sorry for the delay. FYI I've fixed up the CI tests, so now we'll need them green to merge.

I'll try and find some time to review your changes soon!

File: controllers/binding_controller.go [hunk 1]
```diff
@@ -507,19 +508,23 @@ func (r *BindingReconciler) createSecret(instance *ibmcloudv1.Binding, keyConten
...
+			return err
 		}
+		currentBindingInstance.Status.State = bindingStateOnline
+		currentBindingInstance.Status.Message = bindingStateOnline
+		currentBindingInstance.Status.SecretName = getSecretName(currentBindingInstance)
+		return r.Status().Update(context.Background(), currentBindingInstance)
+	})
+	if err != nil {
+		r.Log.Error(err, "Failed to update binding instance after retry", "namespace", instance.Namespace, "name", instance.Name)
+		return ctrl.Result{Requeue: true}, nil
```
Comment (JohnStarich) [inline controllers/binding_controller.go:526]: The switch to retries looks good 👍 
Let's also return the error here
```suggestion
		return ctrl.Result{Requeue: true}, err
```

**Review by JohnStarich — CHANGES_REQUESTED** [review]

**Comment by qibobo** [comment]
@JohnStarich Done. Thank you.

**Review by JohnStarich — COMMENTED** [review]
Code changes look ok 👍 
Please fix the issues reported in Travis, then we can merge

**Comment by qibobo** [comment]
@JohnStarich Fixed the UT, why the codecov is failed?

**Review by JohnStarich — APPROVED** [review]
Looks great, thanks @qibobo! 🎉 

Integration tests passed locally too. ✅

**Pull request merged**